src/rust/engine/process_execution/src/remote.rs

```diff
@@ -433,6 +435,107 @@ impl CommandRunner {
     };
     return stderr;
   }
+
+  fn extract_output_files(
+    &self,
+    execute_response: &bazel_protos::remote_execution::ExecuteResponse,
+  ) -> BoxFuture<Digest, ExecutionError> {
+    let mut futures = vec![];
+    let path_map = Arc::new(Mutex::new(HashMap::new()));
+    let path_map_2 = path_map.clone();
+    let path_stats: Vec<PathStat> = execute_response
+      .get_result()
+      .get_output_files()
+      .into_iter()
+      .map(|output_file| {
+        let output_file_path_buf = PathBuf::from(output_file.get_path());
+        if output_file.has_digest() {
+          let mut underlying_path_map = path_map.lock().unwrap();
+          underlying_path_map.insert(
+            output_file_path_buf.clone(),
+            output_file.get_digest().into(),
+          );
+          PathStat::file(
```

Comment:
This `PathStat` is constructed identically in both branches, so do it outside the `if` rather than twice?